src/Type/Row.purs

```diff
@@ -70,3 +74,17 @@ instance listToRowCons
   :: ( ListToRow tail tailRow
      , RowCons label ty tailRow row )
   => ListToRow (Cons label ty tail) row
+
+
+class Homogeneous (row :: # Type) fieldType
```

Comment:
Also missing a functional dependency here.

Reply:
Good catch.

Reply:
Am I correct in thinking that the right thing to do here is `| row -> fieldType`?